### PR TITLE
🐛 Anchors Not Functioning Correctly for CJK Headings

### PR DESCRIPTION
**Pull request by SmileGuide**
## Issue
While utilizing the theme, it was observed that headings containing non-ASCII characters, such as those in Chinese text, were not positioned when clicked on either the Table of Contents (TOC) or the anchor mark (#) preceding the heading. This issue did not occur with headings in plain English.   
## Resolution  
The resolution involved updating the render-heading.html file to implement URL escaping for anchors, enabling accurate positioning for non-ASCII headings.

### CODE REVIEW

**Comment by nunocoracao** [comment]
Hey @SmileGuide can you give me an example for the bug so that I can test it locally before merging this change?

**Comment by SmileGuide** [comment]
> Hey @SmileGuide can you give me an example for the bug so that I can test it locally before merging this change?  

OK @nunocoracao please take a look at my article <https://www.karlukle.site/blogs/site-build/>  
There are both English and Chinese headings.   
The positioning works fine when clicking on English headings in TOC or their anchors. However, there seems to be an issue when clicking on Chinese headings.
